### DISCUSSION TITLE
Starting from where you left off in the search

### DISCUSSION THREAD

**Discussion by paul188** [MilesCranmer/PySR#616] [Q&A] [answered]
I am currently trying to use Symbolic Regression as part of an optimization routine. 
Now, since I expect the result to be very often very similar from one optimization step to the next, I would like to be able to supply an initial guess or starting point to the Symbolic Regression in order to speed up things. 
Is that already possible ? I could not find any such Option in the EquationSearch.

P.S.: And maybe it would also be possible to suppress printing out the results in each iteration.

**Comment by MilesCranmer** [comment, marked as answer]
You can do warm starts by passing the `saved_state` option to `equation_search`. Make sure to specify `return_state=true` so that `equation_search` actually returns the entire state the first time.

For example:

```julia

saved_state = equation_search(X, y; options=options, return_state=true)

# Continue where it left off:
saved_state = equation_search(X, y; options=options, return_state=true, saved_state=saved_state)
```

However I might recommend using the `MLJ` interface which does this automatically. The second time you run `fit!(mach)`, it will propagate all expressions forward. (And will recompute the losses based on whatever the current dataset is).

Also, in case you were also asking about this, the most relevant other threads:
- https://github.com/MilesCranmer/PySR/discussions/418
- https://github.com/MilesCranmer/PySR/discussions/443

Other threads that are relevant:
- https://github.com/MilesCranmer/PySR/discussions/557
- https://github.com/MilesCranmer/PySR/discussions/465
- https://github.com/MilesCranmer/PySR/discussions/401
- https://github.com/MilesCranmer/PySR/discussions/380
- https://github.com/MilesCranmer/PySR/issues/339
- https://github.com/MilesCranmer/PySR/discussions/417


---

> P.S.: And maybe it would also be possible to suppress printing out the results in each iteration.

You can set this with `verbosity=0`. See https://astroautomata.com/SymbolicRegression.jl/dev/api/ (though there are a lot of options; apologies :))

**Reply by paul188** [reply]
Thanks alot ! That is extremely helpful !
